Packages/com.latios.latiosframework/Mimic/Mecanim/Internal/MecanimInternalUtilities.cs

```diff
@@ -486,32 +486,42 @@ internal static bool Approximately(float a, in float b)
 
                 //If the clip has looped, get a sample of the end of the clip to incorporate it into the delta
                 if (hasLooped)
                 {
                     deltaTransform = clip.SampleBone(0, time);
-                    var previousClipSample = clip.SampleBone(0, time - speedModifiedDeltaTime);
 
-                    var sampleEnd = math.select(clip.duration, -clip.duration, stateSpeed < 0f);
+                    //Change times to account for negative speed
+                    var previousClipSampleTime = math.select(time - speedModifiedDeltaTime, clip.duration + time - speedModifiedDeltaTime, time - speedModifiedDeltaTime < 0f);
+                    var previousClipSample = clip.SampleBone(0, previousClipSampleTime);
+
+                    var sampleEnd = math.select(clip.duration, 0, stateSpeed < 0f);
                     var endClipSample = clip.SampleBone(0, sampleEnd);
 
                     deltaTransform.position += endClipSample.position - previousClipSample.position;
-                    deltaTransform.rotation = math.mul(deltaTransform.rotation, math.mul(math.inverse(endClipSample.rotation), previousClipSample.rotation));
+                    deltaTransform.rotation = math.mul(deltaTransform.rotation, math.mul(endClipSample.rotation, math.inverse(previousClipSample.rotation)));
+                    deltaTransform.scale *= endClipSample.scale / previousClipSample.scale;
+                    deltaTransform.stretch *= endClipSample.stretch / previousClipSample.stretch;
                 }
                 else if (time < clip.duration)
                 {
                     //Get the delta as normal
                     var currentClipSample = clip.SampleBone(0, time);
-                    var previousClipSample = clip.SampleBone(0, time - speedModifiedDeltaTime);
+                    var previousClipSampleTime = math.select(time - speedModifiedDeltaTime, clip.duration + time - speedModifiedDeltaTime, time - speedModifiedDeltaTime < 0f);
+                    var previousClipSample = clip.SampleBone(0, time - previousClipSampleTime);
 
                     deltaTransform.position += currentClipSample.position - previousClipSample.position;
                     deltaTransform.rotation = math.mul(currentClipSample.rotation, math.inverse(previousClipSample.rotation));
+                    deltaTransform.scale *= currentClipSample.scale / previousClipSample.scale;
+                    deltaTransform.stretch *= currentClipSample.stretch / previousClipSample.stretch;
                 }
 
                 currentRoot.position += deltaTransform.position * blendWeight;
                 currentRoot.rotation = math.slerp(currentRoot.rotation, math.mul(currentRoot.rotation, deltaTransform.rotation), blendWeight);
+                currentRoot.scale *= deltaTransform.scale * blendWeight;
+                currentRoot.stretch *= deltaTransform.stretch * blendWeight;
```

Comment:
These still need to be additive with the currentRoot values starting at zero rather than identity. Otherwise the blendWeights scale everything down.

Reply:
I see what you're saying.  Will change and review other scale and stretch assignments.